common/fs/src/cache/watch.rs

```diff
@@ -75,29 +75,27 @@ impl Watcher {
         self.inotify.rm_watch(wd)
     }
 
-    pub fn event_stream<'a>(
-        &mut self,
-        buffer: &'a mut [u8],
-    ) -> std::io::Result<WatchEventStream<'a>> {
+    pub fn event_stream(&mut self) -> std::io::Result<WatchEventStream> {
+        let buffer = vec![0; 4096].into_boxed_slice();
         Ok(WatchEventStream {
             event_stream: self.inotify.event_stream(buffer)?,
         })
     }
 }
 
-pub struct WatchEventStream<'a> {
-    event_stream: inotify::EventStream<&'a mut [u8]>,
+pub struct WatchEventStream {
+    event_stream: inotify::EventStream<Box<[u8]>>,
```

Comment:
Instead of threading this buffer through all the code, it's owned here.

Reply:
I'm not sure why this was ever threaded up, probably a misreading of the type of the inotify.event_stream(buf: T) type signature as requiring a &[u8] rather than as something that can be AsRef'd as [u8]